lib/middleware/handle_bad_request.rb

```diff
@@ -0,0 +1,13 @@
+module Middleware
+  class HandleBadRequest
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      @app.call(env)
+    rescue ActionController::BadRequest => e
```

Comment:
I like this approach but I am a little confused as to how it works. [The Honeybadger](https://app.honeybadger.io/fault/66984/2eb393354a8c34bb594b71ba905d2b69) shows us getting an `ArgumentError` from the code rather than an `ActionController::BadRequest`. If we got a BadRequest error Honeybadger [would actually ignore it itself](https://docs.honeybadger.io/lib/ruby/getting-started/ignoring-errors.html#ignore-by-class) so we shouldn't technically even need Middleware like this. 

Were you able to write a failing spec and get an ArgumentError before implementing this solution?

Reply:
@mstruve Hi, thanks for taking a look!

Yes, the test included was written first and it got me all the way to handling this at the middleware level as the issue does not reach the controller at hand.

I'm attaching a screenshot from my local. You can see there that the error is being caught and handled by the middleware. Also, I discovered and fixed one issue as I learned the `body` needs to respond to `.each`, so it needs to be an array. See: http://programmerscando.com/rails-and-body_proxy-error-nomethoderror/

Screenshot: 
<img width="1168" alt="Screenshot 2020-02-23 at 15 05 05" src="https://user-images.githubusercontent.com/3818442/75113465-7777d000-564e-11ea-9575-78c857f6ae99.png">